tests/test_find_noisy_channels.py

```diff
@@ -1,155 +1,212 @@
 """Test the find_noisy_channels module."""
 import numpy as np
+from numpy.random import RandomState
 import pytest
 
 from pyprep.find_noisy_channels import NoisyChannels
 from pyprep.removeTrend import removeTrend
 
 
-@pytest.mark.usefixtures("raw", "montage")
-def test_findnoisychannels(raw, montage):
-    """Test find noisy channels."""
-    # Set a random state for the test
-    rng = np.random.RandomState(30)
+# Set a fixed random seed for reproducible test results
 
-    raw.set_montage(montage)
-    nd = NoisyChannels(raw, random_state=rng)
-    nd.find_all_bads(ransac=True)
-    bads = nd.get_bads()
-    iterations = (
-        10  # remove any noisy channels by interpolating the bads for 10 iterations
+rng = RandomState(30)
+
+
+# Define some fixtures and utility functions for use across multiple tests
+
+@pytest.fixture(scope="session")
+def raw_clean_detrend(raw_clean):
+    """Return a pre-detrended `mne.io.Raw` object with no bad channels.
+
+    Based on the data from the `raw_clean` fixture, which uses the data for
+    subject 1, run 1 from the Physionet BCI2000 dataset.
+
+    This is only run once per session to save time.
+
+    """
+    raw_clean_detrended = raw_clean.copy()
+    raw_clean_detrended._data = removeTrend(
+        raw_clean.get_data(),
+        raw_clean.info["sfreq"]
     )
-    for iter in range(0, iterations):
-        if len(bads) == 0:
-            continue
-        raw.info["bads"] = bads
-        raw.interpolate_bads()
-        nd = NoisyChannels(raw, random_state=rng)
-        nd.find_all_bads(ransac=True)
-        bads = nd.get_bads()
-
-    # make sure no bad channels exist in the data
-    raw.drop_channels(ch_names=bads)
-
-    # Test for NaN and flat channels
-    raw_tmp = raw.copy()
-    m, n = raw_tmp._data.shape
-    # Insert a nan value for a random channel and make another random channel
-    # completely flat (ones)
-    idxs = rng.choice(np.arange(m), size=2, replace=False)
-    rand_chn_idx1 = idxs[0]
-    rand_chn_idx2 = idxs[1]
-    rand_chn_lab1 = raw_tmp.ch_names[rand_chn_idx1]
-    rand_chn_lab2 = raw_tmp.ch_names[rand_chn_idx2]
-    raw_tmp._data[rand_chn_idx1, n - 1] = np.nan
-    raw_tmp._data[rand_chn_idx2, :] = np.ones(n)
-    nd = NoisyChannels(raw_tmp, random_state=rng)
+    return raw_clean_detrended
+
+
+@pytest.fixture
+def raw_tmp(raw_clean_detrend):
+    """Return an unmodified copy of the `raw_clean_detrend` fixture.
+
+    This is run once per NoisyChannels test, to keep any modifications to
+    `raw_tmp` during a test from affecting `raw_tmp` in any others.
+
+    """
+    raw_tmp = raw_clean_detrend.copy()
+    return raw_tmp
+
+
+def _generate_signal(fmin, fmax, timepoints, fcount=1):
+    """Generate an EEG signal from one or more sine waves in a frequency range."""
+    signal = np.zeros_like(timepoints)
+    for freq in rng.randint(fmin, fmax+1, fcount):
```

Comment:
if you want to use "rng" from outside of the function, instead of passing it into the function (which would probably be cleaner here and throughout), then I think you should make it `RNG` in capital letters, to make it slightly more obvious that it's somewhat like a "constant"

Reply:
Okay, will do.